### DISCUSSION TITLE
GPIO 2 rapid flashing during web OTA update

### DISCUSSION THREAD

**Discussion by ghmpi** [Hieromon/AutoConnect#598] [General]
Hello,

I noticed something curious with the web OTA update function. When updating via the web based update mechanism provided by AutoConnect, GPIO 2 on my ESP32-DEVKITC (Wrover-e) board starts erratically flashing during the upload.

I only notice this because I have a display whose output enable pin is wired to GPIO 2. It doesn't cause any problems but results in the display flashing like a hard drive activity LED when you do a web update!
Using espota.exe to update the board does cause the display to 'hang' but doesn't flash like it does with the web update, it looks like it's just delaying the display update, so I don't think it's the same mechanism that causes the rapid flashing.

It's not causing any problems but I am curious on what may be causing this. Any ideas or where to look?

Thank you!

Regards,
-Moses

**Comment by ghmpi** [comment]
I believe I found it. It is because of the "ticker". I stumbled upon it by accident since it was not easy to know what to search for!
https://hieromon.github.io/AutoConnect/adothers.html#ticker-for-wifi-status

It does not turn off when I specify the following:
config.ticker = false;

But it does stop flickering on GPIO 2 on my board when I specify another GPIO port.
config.tickerPort = 25;

If I am reading it correctly, the documentation seems to imply that even if you specify "config.ticker = false;" it does not turn off the OTA blinking, quote..
"Ticker during OTA
The LED blinking will always be a short blinking during the update via OTA, regardless of the definition of the flicker cycle."

Regards,
-Moses

**Reply by Hieromon** [reply]
@ghmpi LED ticking during OTA is an intentional behavior based on AutoConnect specifications. A description of this feature can be found in the [documentation](http://localhost:8000/AutoConnect/otabrowser.html#updates-with-the-web-browserupdated-wv115). It also has described how to stop port output to GPIO during OTA.
I recommend that you read through all the documentation.

**Comment by ghmpi** [comment]
I will read more of the documentation, thank you for the link.

I will say though.. from reading the statement on the following page... it lead me to believe that you could not modify this behavior.
"Ticker during OTA.. The LED blinking will always be a short blinking during the update via OTA, regardless of the definition of the flicker cycle."
https://hieromon.github.io/AutoConnect/adothers.html#ticker-for-wifi-status

**Reply by Hieromon** [reply]
It is not a descriptive clause about AutoConnect OTA. It means that the ticker is valid even if AutoConnectConfig::ticker is false. That is precisely the phenomenon you mentioned in the [comment](https://github.com/Hieromon/AutoConnect/discussions/598#discussion-5126304).

The conditions for stopping the ticker during **OTA** are described as important in the OTA description.

**Comment by ghmpi** [comment]
I stopped searching about the topic when I saw the sentence with "always" in it. Maybe it's just me and I'm wrong, but it did throw me off for a bit.

**Comment by ghmpi** [comment]
A suggestion.. in additional to the note you may want to mention the capability of turning off the led in the AutoConnectConfig::tickerPort documentation as well.
https://hieromon.github.io/AutoConnect/apiconfig.html#tickerport
